syntax = "proto3";

package suggestions;

// Message representing a single book suggestion.
message BookSuggestion {
  string title = 1;
  string genre = 2;
  string description = 3;
}

// Request message for getting suggestions.
message SuggestionRequest {
  string book_title = 1;
}

// Response message that holds multiple book suggestions.
message SuggestionResponse {
  repeated BookSuggestion suggestions = 1;
}

// Service definition for the suggestion system.
service SuggestionService {
  rpc GetSuggestions(SuggestionRequest) returns (SuggestionResponse);
}